auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/GoogleSignInFragment.kt

```diff
@@ -75,23 +71,58 @@ class GoogleSignInFragment : BaseFragment() {
         updateUI(currentUser)
     }
 
-    private fun handleSignInResult(data: Intent?) {
-        // Result returned from launching the Sign In PendingIntent
-        try {
-            // Google Sign In was successful, authenticate with Firebase
-            val credential = signInClient.getSignInCredentialFromIntent(data)
-            val idToken = credential.googleIdToken
-            if (idToken != null) {
-                Log.d(TAG, "firebaseAuthWithGoogle: ${credential.id}")
-                firebaseAuthWithGoogle(idToken)
-            } else {
-                // Shouldn't happen.
-                Log.d(TAG, "No ID token!")
+    private fun signInWithButton() {
+        // Create the dialog configuration for the Credential Manager request
+        val signInWithGoogleOption = GetSignInWithGoogleOption
+            .Builder(serverClientId = requireContext().getString(R.string.default_web_client_id))
+            .build()
+
+        // Create the Credential Manager request using the configuration created above
+        val request = GetCredentialRequest.Builder()
+            .addCredentialOption(signInWithGoogleOption)
+            .build()
+
+        getCredential(request)
+    }
+
+    private fun signInWithBottomSheet() {
+        // Create the bottom sheet configuration for the Credential Manager request
+        val googleIdOption = GetGoogleIdOption.Builder()
+            .setFilterByAuthorizedAccounts(true)
+            .setServerClientId(requireContext().getString(R.string.default_web_client_id))
+            .build()
+
+        // Create the Credential Manager request using the configuration created above
+        val request = GetCredentialRequest.Builder()
+            .addCredentialOption(googleIdOption)
+            .build()
+
+        getCredential(request)
+    }
+
+    private fun getCredential(request: GetCredentialRequest) {
+        viewLifecycleOwner.lifecycleScope.launch {
+            try {
+                // Launch Credential Manager UI
+                val result = credentialManager.getCredential(
+                    request = request,
+                    context = requireContext()
```

Comment:
In Android it's more common to pass the `Context` first

```suggestion
                    context = requireContext(),
                    request = request                    
```